aggregator_api/src/lib.rs

```diff
@@ -136,87 +140,113 @@ async fn post_task<C: Clock>(
-                    bytes.len()
-                ),
+                format!("invalid role {role}"),
                 Status::BadRequest,
-            ));
+            ))
         }
-        Vec::from([SecretBytes::new(bytes)])
-    } else {
-        Vec::from([SecretBytes::new(
-            thread_rng()
-                .sample_iter(Standard)
-                .take(req.vdaf.verify_key_length())
-                .collect(),
-        )])
     };
-    let task_expiration = req.task_expiration.map(Time::from_seconds_since_epoch);
-    let time_precision = Duration::from_seconds(req.time_precision);
-    let aggregator_auth_tokens = if let Some(encoded) = req.aggregator_auth_token.as_ref() {
-        let token_bytes = URL_SAFE_NO_PAD.decode(encoded).map_err(|err| {
+
+    let vdaf_verify_key_bytes = URL_SAFE_NO_PAD
+        .decode(&req.vdaf_verify_key)
+        .map_err(|err| {
             Error::new(
-                format!("Invalid base64 value for aggregator_auth_token: {err}"),
+                format!("Invalid base64 value for vdaf_verify_key: {err}"),
                 Status::BadRequest,
             )
         })?;
-        Vec::from([
-            // TODO(#472): Each token in the PostTaskReq should indicate whether it is a bearer
-            // token or a DAP-Auth-Token. For now, assume the latter.
-            DapAuthToken::try_from(token_bytes)
-                .map(AuthenticationToken::DapAuth)
-                .map_err(|_| {
-                    Error::new(
-                        "Invalid HTTP header value in aggregator_auth_token".to_string(),
-                        Status::BadRequest,
-                    )
-                })?,
-        ])
-    } else {
-        // TODO(#472): switch to generating bearer tokens by default
-        Vec::from([AuthenticationToken::DapAuth(random())])
-    };
-    let collector_auth_tokens = match (req.role, req.collector_auth_token) {
-        // TODO(#472): switch to generating bearer tokens by default
-        (Role::Leader, None) => Vec::from([AuthenticationToken::DapAuth(random())]),
-        (Role::Leader, Some(encoded)) => {
+    if vdaf_verify_key_bytes.len() != req.vdaf.verify_key_length() {
+        return Err(Error::new(
+            format!(
+                "Wrong VDAF verify key length, expected {}, got {}",
+                req.vdaf.verify_key_length(),
+                vdaf_verify_key_bytes.len()
+            ),
+            Status::BadRequest,
+        ));
+    }
+
+    // DAP recommends deriving the task ID from the VDAF verify key. We deterministically obtain a
+    // 32 byte task ID by taking SHA-256(VDAF verify key).
+    // https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-04#name-verification-key-requiremen
+    let task_id = TaskId::try_from(digest(&SHA256, &vdaf_verify_key_bytes).as_ref())
+        .map_err(|err| Error::new(err.to_string(), Status::BadRequest))?;
+
+    let vdaf_verify_keys = Vec::from([SecretBytes::new(vdaf_verify_key_bytes)]);
```

Comment:
Rust newbie question: what's the preference for `Vec::from([foo])` instead of `vec![foo]`? I would have leaned towards `vec!` but curious if there's something more subtle here.

Reply:
I'm not sure if there's a Rust-wide convention for this, but we've used `Vec::from([T])` in Janus. There's nothing explicit about this in our [code style](https://github.com/divviup/janus/blob/main/docs/DEVELOPMENT.md#code-style), but I remember discussing this in the past -- @branlwyd do you remember if we wrote down anything about this style preference somewhere?